### PR TITLE
[ADD][8.0] stock_analytic

### PR DESCRIPTION
**Pull request by andhit-r**
Move stock_analytic into OCA/account-analytic. The module retrieve from https://github.com/julius-network-solutions/julius-openobject-addons and https://github.com/ClearCorp-dev/stock-logistics-workflow.

Commit history preserved from both branch. Several merge conflict had been resolved.

### CODE REVIEW

File: stock_analytic/README.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+Adds an analytic account in stock move to be able to get analytic information
+when generating the journal items.
+
+Credits
+=======
+
+Contributors
+------------
+
+* Fabio Vílchez <fabio.vilchez@clearcorp.co.cr>
```
Comment (pedrobaeza) [inline stock_analytic/README.rst:69]: Add here Julius people and you

**Comment by pedrobaeza** [comment]
Same comments as in https://github.com/OCA/account-analytic/pull/52

**Comment by andhit-r** [comment]
@pedrobaeza i'm not confident regarding my retrieving technique. Is it safe to proceed?

**Comment by pedrobaeza** [comment]
Yeah, I think so. It seems totally correct.

**Comment by andhit-r** [comment]
@pedrobaeza I noticed that this module depend on **purchase** and **sale** module. Is it wise to depend on those modules since we only need **stock_account** to give analytic account capability to stock.move?

Maybe I could create **stock_sale_analytic**, **stock_purchase_analytic**, and maybe **stock_mrp_analytic** to achieve modularity.

**Comment by pedrobaeza** [comment]
For what is that modules needed? If there's no other choice, to keep modularity it would be good to split it.

**Comment by andhit-r** [comment]
I mean this original module will **force users** to install sale and purchase. I think the main functionality for this module **only to give analytic account on stock.move**. So when stock.move create account.move.line it will automatically fill analytic account field on account.move.line. So no need to depend on sale and purchase.

**stock_sale_analytic**
depend: stock_analytic, sale_stock
functionality: automatically give analytic account on stock.move according to project_id field on sale.order

**stock_purchase_analytic**
depend: purchase, stock_analytic
functionality: automatically give analytic account on stock.move according to account_analytic_id field on purchase.order

I don't like to force users install modules that they don't need. But if my suggestion is somehow overkill I will continue as it is.

**Comment by pedrobaeza** [comment]
No, of course is a good proposition, but I was asking for the reason to need sale or purchase modules. Now I understand that is to propagate the corresponding analytic accounts, but I agree that that should be done in the modules you propose. Maybe a better name should be _sale_stock_analytic_ y _purchase_analytic_ to follow Odoo's naming convention.

**Comment by andhit-r** [comment]
Ok @pedrobaeza. Thanks for your input. I will finish this module first.

**Comment by andhit-r** [comment]
Travis didn't run again

**Comment by pedrobaeza** [comment]
Seeing https://www.traviscistatus.com/, they experimented a problem and now they have some troubles to get up to date with all the checks, so wait a bit or try again to resubmit.

**Comment by andhit-r** [comment]
close to restart travis

**Comment by andhit-r** [comment]
@pedrobaeza would you kindly give need review label on this PR. Thanks before.

**Comment by pedrobaeza** [comment]
Done

**Comment by andhit-r** [comment]
Thanks @pedrobaeza.

**Comment by andhit-r** [comment]
Hi @OCA/accounting-maintainers would you kindly review this PR. Many thanks

**Comment by pedrobaeza** [comment]
:+1:

**Comment by andhit-r** [comment]
Thanks @pedrobaeza

**Comment by andhit-r** [comment]
Hi @OCA/accounting-maintainers would you kindly review this PR. Many thanks

**Comment by JordiBForgeFlow** [comment]
@aheficent and me will review.

**Comment by andhit-r** [comment]
Thank you @jbeficent

**Comment by aheficent** [comment]
IMHO I think It should create the analytic entries for income or expenses account only. It seems to me that it's creating always the analytic entries for any accounting entry: https://github.com/andhit-r/account-analytic/blob/8.0-stock_analytic_1/stock_analytic/models/stock.py#L36 is creating. Do you agree with me @andhit-r ?
cc @jbeficent

File: stock_analytic/models/stock.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        res = super(StockQuant,
+                    self)._prepare_account_move_line(
+                        move, qty, cost,
+                        credit_account_id,
+                        debit_account_id,
+                        context=context
+                        )
+
+        # Add analytic account in debit line
+        if move.account_analytic_id:
```
Comment (aheficent) [inline stock_analytic/models/stock.py]: IMHO I think It should create the analytic entries for income or expenses account only. This creates two analytic entries, one for the debit and one for the credit, even for assets accounts. Do you agree with me @andhit-r ?
cc @jbeficent

File: stock_analytic/models/stock.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            res[0][2].update({
+                'analytic_account_id': move.account_analytic_id.id,
+            })
+            res[1][2].update({
+                'analytic_account_id': move.account_analytic_id.id,
+            })
+        return res
```
Comment (AaronHForgeFlow) [inline stock_analytic/models/stock.py:45]: IMHO I think It should create the analytic entries for income or expenses account only. This creates two analytic entries, one for the debit and one for the credit, even for assets accounts. Do you agree with me @andhit-r ?
cc @jbeficent

**Comment by AaronHForgeFlow** [comment]
What else is needed to get this merged?

**Comment by pedrobaeza** [comment]
Get more reviewers. You haven't approved the PR yet and there's a question unanswered as I see.

File: stock_analytic/models/stock.py (hunk 3 above)
Comment (andhit-r) [reply stock_analytic/models/stock.py:45]: @aheficent agree. I push a commit to correct this. Analytic account will only be assigned if account != valuation account

Comment (andhit-r) [reply stock_analytic/models/stock.py:45]: Sorry for really-really late reply. Kindly review this again. Many thanks.

**Comment by andhit-r** [comment]
Try to restart travis

**Comment by andhit-r** [comment]
Travis fails because of other modules

**Comment by andhit-r** [comment]
@jbeficent @aheficent all checks green. Would you kindly review it again. Many thanks.

@mikevhe18 @azmimr67 please help review

File: stock_analytic/models/stock.py (hunk 3 above)
Comment (AaronHForgeFlow) [reply stock_analytic/models/stock.py:45]: Thank you @andhit-r I like the proposed changes. There's a problem with runbot, so i will on local. The code looks correct to me.

File: stock_analytic/models/stock.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+# © 2016 Andhitia Rama <andhitia.r@gmail.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import fields, models, api
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    account_analytic_id = fields.Many2one(
```
Comment (AaronHForgeFlow) [inline stock_analytic/models/stock.py]: IMHO i think that the name of the field should be analytic_account_id. I think the name of the field should match the name of other modules as for example the account module https://github.com/OCA/OCB/blob/8.0/addons/account/account.py#L2398 or the project module https://github.com/OCA/OCB/blob/8.0/addons/project/project.py#L231

**Review by AaronHForgeFlow — COMMENTED** [review]
I tested on local and reviewed the code and it works correctly. The only issue is related with the name of the field. What do you think about this?

**Review by AaronHForgeFlow — APPROVED** [review]
Tested on runbot + code review :+1: 

Thank you @andhit-r and @mikevhe18 Great job!

**Comment by mikevhe18** [comment]
:+1: tested on runbot

**Pull request merged**